### PR TITLE
Implemented DELETE for existing skills

### LINKED ISSUE #3: Delete existing Skill
Using a `DELETE` request for the `/resume/skill` route, delete an existing Skill using its index position as an ID. As part of this, you should write a test in `test_pytest.py` to show it works.

### PR DESCRIPTION
**Pull request by nfallah**
## Summary
Added and updated the `/resume/skill` endpoint for DELETE requests to remove existing skills. On success, the erased skill is returned in JSON format, and both `data` in memory and `data.json` on disk are updated.

## Issue
This request closes issue #3.

## Testing
An example DELETE request for `/resume/skill?index=value` will succeed if a valid JSON is provided and `value` is numeric and within bounds. The provided test case `test_skill_delete()` in `test_pytest.py` calls this endpoint with index 0. It then validates the removal by calling a GET request at the same index, ensuring an error or a different skill is returned.

## Concerns
This implementation does not use the serialization class we discussed earlier as the issue has not yet been resolved. Similarly, `index` is obtained via query (not URL) parameters, which is another concern our team has discussed. Lastly, there can be mismatches between the `id` attribute and the actual index of a given skill; deleting a skill will decrement the indices of all skills in front, but not their ids. As such, the `id` of different skills can be identical under certain conditions, and may generally lead to problems.

### CODE REVIEW

File: test_pytest.py [hunk 1]
```diff
@@ -108,3 +108,15 @@ def test_skill():
 
     response = app.test_client().get('/resume/skill')
     assert response.json[item_id] == example_skill
+
+def test_skill_delete():
```
Comment (Pradyuman7) [inline test_pytest.py:112]: Same thing, does this test pass atm?

Comment (nfallah) [reply test_pytest.py:112]: Yes, the test does currently pass.

Comment (Pradyuman7) [reply test_pytest.py:112]: In the pipeline it looks like this fails. The issue doesn't seem to be the content of the test, but the fact that there is now an id field in the skill model. Maybe you can add the id field in this test, as I think we have another issue that tracks what we should for the id addition

Comment (Pradyuman7) [reply test_pytest.py:112]: Oops, my bad! I was looking at a different test

**Review by Pradyuman7 — APPROVED** [review]
LGTM, please just resolve the conflicts

**Pull request merged**